services/venice-controller/src/main/java/com/linkedin/venice/controller/init/ControllerSystemSchemaInitializer.java

```diff
@@ -0,0 +1,160 @@
+    if (admin.isLeaderControllerFor(multiClusterConfigs.getSystemSchemaClusterName())) {
+      registerSystemSchemaAsLeaderController(systemStore, updateStoreParams);
+    } else if (multiClusterConfigs.getCommonConfig().shouldAutoRegisterWriteSystemSchemas()) {
+      registerSystemSchemaUsingControllerClient(systemStore, updateStoreParams);
+    } else {
+      throw new VeniceUnsupportedOperationException("Schema system store initialization for non-leader controllers");
+    }
+  }
+
+  private void registerSystemSchemaAsLeaderController(VeniceSystemStoreType systemStore) {
+    new LeaderControllerSystemSchemaInitializer(systemStore, multiClusterConfigs, admin).execute();
+  }
+
+  private void registerSystemSchemaAsLeaderController(
+      VeniceSystemStoreType systemStore,
+      UpdateStoreQueryParams updateStoreParams) {
+    new LeaderControllerSystemSchemaInitializer(systemStore, multiClusterConfigs, admin, updateStoreParams, true)
+        .execute();
+  }
+
+  private void registerSystemSchemaUsingControllerClient(VeniceSystemStoreType systemStore) {
+    new ControllerClientBackedSystemSchemaInitializer(
+        parentControllerClient.get(),
+        systemStore,
+        new SchemaPresenceChecker(
+            new SchemaRepoBackedSchemaReader(schemaRepo, systemStore.getZkSharedStoreName()),
+            systemStore.getValueSchemaProtocol()),
+        multiClusterConfigs.getCommonConfig().shouldAutoCreateWriteSystemStores()).execute();
+  }
+
+  private void registerSystemSchemaUsingControllerClient(
+      VeniceSystemStoreType systemStore,
+      UpdateStoreQueryParams updateStoreParams) {
+    new ControllerClientBackedSystemSchemaInitializer(
+        parentControllerClient.get(),
+        systemStore,
+        new SchemaPresenceChecker(
+            new SchemaRepoBackedSchemaReader(schemaRepo, systemStore.getZkSharedStoreName()),
+            systemStore.getValueSchemaProtocol()),
+        multiClusterConfigs.getCommonConfig().shouldAutoCreateWriteSystemStores(),
+        updateStoreParams,
+        true).execute();
+  }
+
+  @Override
+  public void close() {
+    parentControllerClient.ifPresent(Utils::closeQuietlyWithErrorLogged);
+    parentColoD2Client.ifPresent(D2ClientUtils::shutdownClient);
+  }
+
+  @Override
+  public void execute(String clusterToInit) {
+    registerSystemSchema(VeniceSystemStoreType.KAFKA_MESSAGE_ENVELOPE);
+    registerSystemSchema(VeniceSystemStoreType.PARTITION_STATE);
+    registerSystemSchema(VeniceSystemStoreType.STORE_VERSION_STATE);
+
+    if (multiClusterConfigs.isZkSharedMetaSystemSchemaStoreAutoCreationEnabled()) {
+      // Add routine to create zk shared metadata system store
+      UpdateStoreQueryParams metadataSystemStoreUpdate =
+          new UpdateStoreQueryParams().setHybridRewindSeconds(TimeUnit.DAYS.toSeconds(1)) // 1 day rewind
+              .setHybridOffsetLagThreshold(1)
+              .setHybridTimeLagThreshold(TimeUnit.MINUTES.toSeconds(1)) // 1 mins
```

Comment:
This and line153 should be set to -1 to disable time lag measurement on system store
```
.setHybridTimeLagThreshold(-1) // Explicitly disable hybrid time lag measurement on system store
```

Reply:
You're right. Wonder where I got this. Will update and audit all such configs

Reply:
Turns out this was the previous config. Updated to match the latest configs